kubernetes-client/src/main/java/io/fabric8/kubernetes/client/CustomResource.java

```diff
@@ -71,11 +65,10 @@
   "status"
 })
 @Buildable(builderPackage = "io.fabric8.kubernetes.api.builder", editableEnabled = false)
-public abstract class CustomResource<S, T> implements HasMetadata {
+public abstract class CustomResource<S, T> implements HasMetadata, CustomResourceMetadata<S, T> {
```

Comment:
If the CustomResourceMetadata extends HasMetadata, it should not be necessary to duplicate HasMetadata here.